src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts

```diff
@@ -0,0 +1,367 @@
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  forwardRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  Optional,
+  Renderer2
+} from '@angular/core';
+
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  Validator,
+  ValidationErrors
+} from '@angular/forms';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import 'rxjs/add/operator/distinctUntilChanged';
+
+import 'rxjs/add/operator/takeUntil';
+
+import {
+  SkyDatepickerConfigService
+} from './datepicker-config.service';
+
+import {
+  SkyDatepickerComponent
+} from './datepicker.component';
+import { SkyFuzzyDate } from './fuzzy-date';
+import { SkyFuzzyDateService } from './fuzzy-date.service';
+import { SkyDatepickerInputDirective } from './datepicker-input.directive';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_DATEPICKER_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyFuzzyDatepickerInputDirective),
+  multi: true
+};
+
+const SKY_DATEPICKER_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyFuzzyDatepickerInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyFuzzyDatepickerInput]',
+  providers: [
+    SKY_DATEPICKER_VALUE_ACCESSOR,
+    SKY_DATEPICKER_VALIDATOR
+  ]
+})
+export class SkyFuzzyDatepickerInputDirective extends SkyDatepickerInputDirective
+  implements OnInit, OnDestroy, AfterViewInit, AfterContentInit, ControlValueAccessor, Validator {
+
+  private _yearRequired: boolean = false;
+
+  @Input()
+  public set yearRequired(value: boolean) {
+    this._yearRequired = value;
+    this.onValidatorChange();
+  }
+
+  public get yearRequired(): boolean {
+    return this._yearRequired;
+  }
+
+  private _cannotBeFuture: boolean = false;
+
+  @Input()
+  public set cannotBeFuture(value: boolean) {
+    this._cannotBeFuture = value;
+    this.onValidatorChange();
+  }
+
+  public get cannotBeFuture(): boolean {
+    return this._cannotBeFuture;
+  }
+
+  private _maxFuzzyDate: SkyFuzzyDate;
+
+  @Input()
+  public set maxFuzzyDate(value: SkyFuzzyDate) {
```

Comment:
I'm a little concerned about the overlap of `SkyFuzzyDatepickerInputDirective` and `SkyDatepickerInputDirective` here. Thinking about this from a docs perspective, which inherited input properties from the parent directive will consumers need to use vs not use? For example, they shouldn't be setting `maxDate` and `minDate`, but instead setting `minFuzzyDate ` and `maxFuzzyDate `. If the docs need to explain that, it may become more complicated than necessary.

I do understand that by extending the `SkyDatepickerInputDirective`, you got a lot of stuff for free, but I'm wondering if it complicates things a little more and that outweighs the benefits.

How much would we lose by decoupling the two?

Reply:
I agree we should separate the two directives if possible. Also, how would a min/max date work with a fuzzy date?

Reply:
Following a discussion, we've decided to leave `SkyFuzzyDatepickerInputDirective` extending `SkyDatepickerInputDirective`